x/time.go

```diff
@@ -9,6 +9,8 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+var rnd = rand.New(rand.NewSource(time.Now().Unix()))
```

Comment:
@mattbonnell the default rand source has always the same seed (1) which makes the delay pattern predictable.